lib/Generate.js

```diff
@@ -190,39 +144,24 @@ function generateMainModule(
     moduleName: string,
     possiblyTests: Array<string>,
   }> */,
-  generatedSrc /*: string */,
+  mainModule /*: { moduleName: string, path: string } */,
   processes /*: number */
-) /*: string */ {
+) /*: void */ {
   const testFileBody = makeTestFileBody(
     testModules,
     makeOptsCode(fuzz, seed, report, testFileGlobs, testFilePaths, processes)
   );
 
-  // Generate a filename that incorporates the hash of file contents.
-  // This way, if you run e.g. `elm-test Foo.elm` and then `elm-test Bar.elm`
-  // and then re-run `elm-test Foo.elm` we still have a cached `Main` for
-  // `Foo.elm` (assuming none of its necessary imports have changed - and
-  // why would they?) so we don't have to recompile it.
-  const salt = Murmur.murmur3(testFileBody);
-  const moduleName = 'Main' + salt;
-  const mainPath = path.join(generatedSrc, 'Test', 'Generated');
-  const mainFile = path.join(mainPath, moduleName + '.elm');
-
-  // We'll be putting the generated Main in something like this:
-  //
-  // my-project-name/elm-stuff/generated-code/elm-community/elm-test/src/Test/Generated/Main123456.elm
-  const testFileContents = `module Test.Generated.${moduleName} exposing (main)\n\n${testFileBody}`;
+  const testFileContents = `module ${mainModule.moduleName} exposing (main)\n\n${testFileBody}`;
 
-  // Make sure src/Test/Generated/ exists so we can write the file there.
-  fs.mkdirSync(mainPath, { recursive: true });
+  // Make sure the generated code dir exists so we can write the file there.
+  fs.mkdirSync(path.dirname(mainModule.path), { recursive: true });
 
   // Always write the file, in order to update its timestamp. This is important,
-  // because if we run `elm-make Main123456.elm` and that file's timestamp did
-  // not change, elm-make will short-circuit and not recompile *anything* - even
+  // because if we run `elm make Main.elm` and that file's timestamp did
+  // not change, `elm make` will short-circuit and not recompile *anything* - even
   // if some of Main's dependencies (such as an individual test file) changed.
```

Comment:
Is this still true? I think the elm compiler is now smarter than this.

Reply:
You seem to be right! I can’t reproduce this problem.

I’ve also tested watch mode a bit at my work code base, and always generating the same file name (Main.elm) seems to cause no issues.

I removed the comment.

Reply:
Some of the test runner seems really old. Is so nice to get rid of this tech debt.